lib/monads/monad.rb

```diff
@@ -12,6 +12,12 @@ def method_missing(*args, &block)
       end
     end
 
+    def respond_to_missing?(method_name, include_private=false)
+      super || within do |value|
```

Comment:
As a matter of principle, I think we this should be `… || super` rather than `super || …`. We should give the monad a chance to reply first, and only ask the superclass if necessary, rather than vice versa. (This probably doesn’t make a difference in practice — I’m thinking about educational value.)